### ISSUE TITLE
Themes

### ISSUE THREAD

**Issue by JWCook** [joshuawootonn/type-the-word#8]
Would you consider adding support for themes? Either built-in themes (with an easy way to submit more via a PR here) or user-defined color settings would be a welcome addition.

For example, monkeytype stores its built-in themes as just 4 color values [stored in a JSON file](https://github.com/monkeytypegame/monkeytype/blob/master/frontend/static/themes/_list.json), and also has a [section in the settings menu](https://github.com/joshuawootonn/type-the-word/assets/419936/f835f23d-f985-4d3f-a9cd-90eeb1919745) to set user-defined colors.

**Comment by joshuawootonn** [comment]
Hey @JWCook! I like this idea a lot. Especially creating a pattern so that other users can contribute. 

I don't have a public roadmap right now, but I am working on a couple of other tasks. I'm [migrating to App router](https://github.com/joshuawootonn/type-the-word/pull/7), and after that I would like to:

- Remember the last typing location [https://github.com/joshuawootonn/type-the-word/issues/5](https://github.com/joshuawootonn/type-the-word/issues/5)
- Make the entire `/history` section more helpful
- And add a leader board

Before those last three tasks I will be reaching out to users to see what they would like most and I'll throw this one into the mix.

**Comment by joshuawootonn** [comment]
Seems like most of your work is in python, but in the off chance you want to contribute I would be happy to help design a theme'ing setup like you described. Let me know!

**Comment by JWCook** [comment]
Sounds good! I occasionally do a little frontend work, but I may not have enough experience with this stack to be of much help.

I did look around a bit to see how others handle multiple themes with tailwind, though. There are several plugins out there that aim to make it easier, and two of the more actively maintained ones appear to be [tw-colors](https://github.com/L-Blondy/tw-colors) and [tailwindcss-themer](https://github.com/RyanClementsHax/tailwindcss-themer).

From a quick look through this repo, it looks like the background and text colors currently come from builtin classes (`bg-white`, `text-black`, etc.) applied directly to each element, so I suppose the first step would be to move all that to [tailwind theme config](https://tailwindcss.com/docs/theme), then try out one of those plugins to support multiple themes. That's just one idea, though; there might be other ways to go about it, like CSS variables.

**Comment by joshuawootonn** [comment]
> I suppose the first step would be to move all that to [tailwind theme config](https://tailwindcss.com/docs/theme),

Definitely. I think this is the first step. Thankfully the themes I am using are so simple that it won't be to complicated. I really like [how shadcn is doing this](https://ui.shadcn.com/docs/theming#css-variables). So I would like to follow that convention of variables.

**Comment by joshuawootonn** [comment]
> Sounds good! I occasionally do a little frontend work, but I may not have enough experience with this stack to be of much help.

No worries. I'll get to it at some point. 

I'll take a look at those two plugins too. I haven't seen those yet.

**Comment by joshuawootonn** [comment]
@JWCook Hey I made the themes configurable. You can now toggle between light and dark mode. 

![CleanShot 2024-08-25 at 17 20 57@2x](https://github.com/user-attachments/assets/2c0e9219-7bda-48a2-9ea1-154e0c8a87f0)

I am thinking that next month I'll be able to implement the UI for a custom theme so that you can use custom colors. Stay tuned for that change :)

**Comment by JWCook** [comment]
Great, thanks for the update! Just tested it out and it looks good.

**Comment by joshuawootonn** [comment]
Hey @JWCook you can now create your own themes in Type the Word. I did this across these PRs in case you are curious how it's implemented.

#12, #14, #15, #16, and #17 

There are a bunch of videos in the [changelog](https://www.typetheword.site/changelog) with details, but just wanted to keep you updated

**Comment by JWCook** [comment]
Fantastic, thank you! I just tried it out, and this is a nice improvement.

In case anyone else wants a [gruvbox](https://github.com/morhetz/gruvbox?tab=readme-ov-file)-like theme:
```
Primary:   251,241,199
Secondary: 80,73,69
Success:   250,189,47
Error:     251,73,52
```

![image](https://github.com/user-attachments/assets/16411727-2f57-4ac8-9ed9-9e55c78d019c)

**Comment by joshuawootonn** [comment]
Sick!
